samza-core/src/main/java/org/apache/samza/operators/impl/SendToTableOperatorImpl.java

```diff
@@ -54,11 +62,10 @@ protected void handleInit(Context context) {
   }
 
   @Override
-  protected CompletionStage<Collection<Void>> handleMessageAsync(KV<K, V> message, MessageCollector collector,
+  protected CompletionStage<Collection<KV<K, V>>> handleMessageAsync(KV<K, V> message, MessageCollector collector,
       TaskCoordinator coordinator) {
     table.put(message.getKey(), message.getValue());
-    // there should be no further chained operators since this is a terminal operator.
-    return CompletableFuture.completedFuture(Collections.emptyList());
+    return CompletableFuture.completedFuture(new ArrayList<KV<K, V>>(Arrays.asList(message)));
```

Comment:
Do you need to wrap Arrays.asList in the ArrayList?

Reply:
is it sufficient to use ImmutableList

Reply:
+1. `Collections.singleton(message)`